README.md

```diff
@@ -31,142 +31,225 @@ The game is a space shooter with a super overly dramatic / cheesy 80s style bada
+
+Extension classes should be named like ______Extensions.  This gets weird though.  If you're extending the string class you should name it StringExtensions, but for Vector3 / Vector2 its okay to just put the code inside VectorExtensions because Vector3 is a sub class of Vector2.
+
+> Avoid Magic Strings and Magic Numbers
+
+For things like UI and dialogue, **NEVER USE HARD CODED STRINGS**.  Strings such as these need to be added to strings xmls in case we ever wish to localize our game.  
+
+For references, etc. prefer to use a private constant string, even if the string is only used once.  The memory footprint is the same, and it keeps things readable.
+
+Same thing with integers or floats.  I'd rather see a constant at the top saying PLAYER_START_HEALTH so i don't need to sift through code to see what the player start health is.
+
+### Components
+
+> Every single "component" class should be somewhere within the "Components" folder, and everything within the "Components" folder should be a component class
+
+A **component class** is a class that can be dragged and dropped onto game objects in Unity.  Having all of these classes in this folder makes it beautifully obvious what can be dragged onto objects in unity.
+
+The "base folders" rule still applies to the component classes too though.  We can have a "base" folder inside a component class folder, or whatever, as long as we're not dinguses who try to drag a base class onto an object.
+
+In our case, these classes will almost always inherit from ```ComponentBase```.  Component Base is a class that sits between our components and the Unity "MonoBehaviour" class in the inheritance structure. 
+
+**Random Pro tip**: Be careful creating component scripts via unity (rather than visual studio).  Unity doesn't add namespaces, so you'll have to add one manually.
+
+#### ComponentBase
+
+```ComponentBase``` gives us a few main benefits:
+ 
+* It allows us to write global helper methods.  These include things like .```GetRequiredComponent<T>``` which automatically performs null checking if we expect a component to exist on an object.
+* It gives us the ability to control every single component on a global level.  Things such as global exception handling or game speed manipulation are now much easier.  
+* Overriding values for Step, create, etc. is much easier than remembering the arbitrary method names "Update", "Activate", etc.
+
+Basically, every single component should inherit from ComponentBase.  The one exception we have so far is for the ```SystemObjectBehavior```, which can't because it gets created *from* ComponentBase which causes infinite recursion.
+
+```ComponentBase``` has a few overridable methods: 
+
+* **Construct**: This is the closest thing to a constructor we are allowed to use in Unity.  Here we set up references, etc.  At this point in code, the structure of everything is set up but things aren't rendered yet. (Unity's equivalent of Awake())
```

Comment:
Why is this all jBleu? Not sure if there is an extra back tick or something but formatting looks weird with the coloring.

Reply:
idk fam but it looks gucci on the git hub page